doc/gettext.rst

```diff
@@ -0,0 +1,379 @@
+
+* ``dgettext``, ``_d``
+
+  Translation in ``LC_MESSAGES`` category and specified domain.
+
+  .. code-block:: jinja
+  
+      {{ 'String'|dgettext('domain') }}
+      {{ 'String'|_d('domain') }}
+      {{ dgettext('domain', 'String') }}
+      {{ _d('domain', 'String') }}
+
+* ``dpgettext``, ``_dp``
+
+  Translation in ``LC_MESSAGES`` category and specified domain with context.
+
+  .. code-block:: jinja
+  
+      {{ 'String'|dpgettext('domain', 'context') }}
+      {{ 'String'|_dp('domain', 'context') }}
+      {{ dpgettext('context', 'domain', 'String') }}
+      {{ _dp('context', 'domain', 'String') }}
+
+* ``dngettext``, ``_dn``
+
+  Pluralized translation in ``LC_MESSAGES`` category and specified domain.
+
+  .. code-block:: jinja
+  
+      {{ 'Singular'|dngettext('Plural', num, 'domain') }}
+      {{ 'Singular'|_dn('Plural', num, 'domain') }}
+      {{ dngettext('domain', 'Singular', 'Plural', num) }}
+      {{ _dn('domain', 'Singular', 'Plural', num) }}
+
+* ``dnpgettext``, ``_dnp``
+
+  Pluralized translation in ``LC_MESSAGES`` category and specified domain with context.
+
+  .. code-block:: jinja
+  
+      {{ 'Singular'|dnpgettext('Plural', num, 'domain', 'context') }}
+      {{ 'Singular'|_dnp('Plural', num, 'domain', 'context') }}
+      {{ dnpgettext('context', 'domain', 'Singular', 'Plural', num) }}
+      {{ _dnp('context', 'domain', 'Singular', 'Plural', num) }}
+
+* ``dcgettext``, ``_dc``
+
+  Translation in specified domain and category.
+
+  .. code-block:: jinja
+  
+      {{ 'String'|dcgettext('domain', 'category') }}
+      {{ 'String'|_dc('domain', 'category') }}
+      {{ dcgettext('domain', 'String', 'category') }}
+      {{ _dc('domain', 'String', 'category') }}
+
+* ``dcpgettext``, ``_dcp``
+
+  Translation in specified domain and category with context.
+
+  .. code-block:: jinja
+  
+      {{ 'String'|dcpgettext('domain', 'category', 'context') }}
+      {{ 'String'|_dcp('domain', 'category', 'context') }}
+      {{ dcpgettext('context', 'domain', 'String', 'category') }}
+      {{ _dcp('context', 'domain', 'String', 'category') }}
+
+* ``dcngettext``, ``_dcn``
+
+  Pluralized translation in specified domain and category.
+
+  .. code-block:: jinja
+  
+      {{ 'Singular'|dcngettext('Plural', num, 'domain', 'category') }}
+      {{ 'Singular'|_dcn('Plural', num, 'domain', 'category') }}
+      {{ dcngettext('domain', 'Singular', 'Plural', num, 'category') }}
+      {{ _dcn('domain', 'Singular', 'Plural', num, 'category') }}
+
+* ``dcnpgettext``, ``_dcnp``
+
+  Pluralized translation in specified domain and category with context.
+
+  .. code-block:: jinja
+
+      {{ 'Singular'|dcnpgettext('Plural', num, 'domain', 'category', 'context') }}
+      {{ 'Singular'|_dcnp('Plural', num, 'domain', 'category', 'context') }}
+      {{ dcnpgettext('context', 'domain', 'Singular', 'Plural', 'category') }}
+      {{ _dcnp('context', 'domain', 'Singular', 'Plural', 'category') }}
+
+
+Workflow
```

Comment:
I don't think this belongs to the Twig-extensions documentation. This is about documenting a gettext-based workflow. If someone decides to use the gettext extension for Twig, they should already know how a gettext workflow works (or look for a blog post explaining it)